test/ERC6538Registry.t.sol

```diff
@@ -56,15 +55,215 @@ contract RegisterKeys is ERC6538RegistryTest {
+      emit StealthMetaAddressSet(alice, schemeId, stealthMetaAddress);
+      registry.registerKeysOnBehalf(alice, schemeId, signature, stealthMetaAddress);
+    }
+  }
+
+  function testFuzz_RevertIf_SignatureIsNotValid(
+    string memory name,
+    address bob,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+    bytes32 hash = keccak256(abi.encode(alice, schemeId, stealthMetaAddress, 0));
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectRevert("Invalid signature");
+    registry.registerKeysOnBehalf(bob, schemeId, signature, stealthMetaAddress);
+  }
+
+  function testFuzz_RevertIf_WrongNonce(
+    string memory name,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress,
+    uint256 nonce
+  ) external {
+    vm.assume(nonce != 0);
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+    bytes32 hash = keccak256(abi.encode(alice, schemeId, stealthMetaAddress, nonce));
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectRevert("Invalid signature");
+    registry.registerKeysOnBehalf(alice, schemeId, signature, stealthMetaAddress);
+  }
+
+  function test_RevertIf_NoSignatureIsProvided() external {
+    vm.expectRevert("Invalid signature");
     registry.registerKeysOnBehalf(address(0), 0, "", "");
   }
 }
 
-contract RegisterKeysOnBehalf_Bytes is ERC6538RegistryTest {
-  function test_NotImplemented() external {
-    vm.expectRevert("not implemented");
-    registry.registerKeysOnBehalf(bytes("0"), 0, "", "");
+contract ERC1271MockContract {
+  address public owner;
+
+  enum RecoverError {
+    NoError,
+    InvalidSignature,
+    InvalidSignatureLength,
+    InvalidSignatureS
+  }
+
+  error ECDSAInvalidSignature();
+  error ECDSAInvalidSignatureLength(uint256 length);
+  error ECDSAInvalidSignatureS(bytes32 s);
+
+  constructor() {
+    owner = msg.sender;
+  }
+
+  function isValidSignature(bytes32 hash, bytes memory signature)
+    public
+    view
+    returns (bytes4 magicValue)
+  {
+    return recover(hash, signature) == owner ? this.isValidSignature.selector : bytes4(0);
+  }
+
+  function recover(bytes32 hash, bytes memory signature) internal pure returns (address) {
+    (address recovered, RecoverError error, bytes32 errorArg) = tryRecover(hash, signature);
+    _throwError(error, errorArg);
+    return recovered;
+  }
+
+  function tryRecover(bytes32 hash, bytes memory signature)
+    internal
+    pure
+    returns (address, RecoverError, bytes32)
+  {
+    if (signature.length == 65) {
+      bytes32 r;
+      bytes32 s;
+      uint8 v;
+      // ecrecover takes the signature parameters, and the only way to get them
+      // currently is to use assembly.
+      /// @solidity memory-safe-assembly
+      assembly {
```

Comment:
nit, the comment approach is nice for backwards compatibility with older solc versions (e.g. if you're building a library intended to be consumed by devs), but since these files have a hardcoded pragma and we can enforce a minimum pragma, you can use the other version https://docs.soliditylang.org/en/latest/assembly.html
```suggestion
      assembly ("memory-safe") {
```